EasyCommands/CommandParsers/ParameterProcessors.cs

```diff
@@ -136,7 +136,11 @@ public class PrimitiveProcessor<T> : ParameterProcessor<T> where T : class, Prim
 
                 Variable variable = new AmbiguousStringVariable(value.value);
 
-                if (primitive != null || value.isExplicit) variable = new StaticVariable(primitive ?? ResolvePrimitive(value.value));
+                if (primitive != null || value.isExplicit) {
+                    variable = new StaticVariable(primitive ?? ResolvePrimitive(value.value));
+                    if (variable.GetValue().returnType == Return.VECTOR)
+                        variable = new VectorVariable(variable);
+                }
```

Comment:
I don't think we need to adjust this one.  This is only in place because we are trying to use VectorVariables for all instances of using Vector, and I don't think that's necessary.

I think we de-scope "VectorVariable" to only be used if you are creating a Vector via variable:variable:variable.

The case of float:float:float can be left alone.  It's ok that vectors of this form aren't processed as primitives but rather get processed as a dynamic type of variable later by the parsing rule.

Note this also great simplifies the code needed in VectorVariable

Reply:
This probably makes GPS parsing challenging, as you likely found out and what also may have led you to try and combine these.  

Perhaps the answer is that we stop trying to parse vectors at all inside of ParsePrimitive, insteaad relying only on Processing Rules.  Floating point numbers will still get resolved to static variables, at which point they will then get parsed into either a GPS variable (var:var:var:var:var), otherwise a vector (var:var:var), otherwise possibly a ternary (var?var:var)

Reply:
* You can try the first commit to see what happens if you remove the parsing of _primitive_ vectors. Several of the test fail because they rely either on the direct parsing in the tokenizing step or on line 140 above. Also #170 is in full effect then!

* Unfortunatly i found that i had to wrap every vector into a `VectorVariable` so i can distinguish `VectorVariable`s from other Variables. This is necessary otherwise the vector rule is to greedy, see discussion below.

* The snippet above only affects _primitive_ vectors already, all other cases are handled by the new vector rule.

* What form should that even take? I still can not understand why `ParseVectorFromGPSCoordinate` even works right now.

* First see the first point. Secondly `var ? var:var:var : var:var:var`, `var? var : var:var:var:var:var` and `var? var:var:var:var:var : var` cannot be distinguished __even if__ we find a way to define vector and gps rules that not so greedy to consume themself and eachother again (see discusion below).

Reply:
This isn't the first instance of a command line being potentially ambiguous, and it won't be the last unfortunately.

I think we can make it clear in the docs that  ```var?var:var:var``` is always parsed as var ? (var:var:var) by default.  A user can clarify their intent, if needed, using parentheses: ```(var ? var : var) : var : var```, or ```var ? (var:var:var) : var```.  

We could change the vector dilimeter, true, though I definitely prefer it, especially given that it is the delimiter used by Space Engineers for GPS coords.  I think we want to keep the ability to directly parse a GPS coordinate for most cases.

I think we fix the critical parsing bug first, and then remove vector parsing entirely from Primitive Parsing, relying on the rules instead.  And we make it clear in the docs on both Ternary Operation and Vector Creation that if you use the two together the default behavior is used and to use parentheses if needed.

I apologize, I haven't had any time to sit down and look at some of this in detail; I should have some time tomorrow night to really look at this.

Reply:
Given the default behavior, I don't think the check against VectorVariable would be needed anymore.  The rule would simply be "if you see var : var : var, that's a vector.  And since we parse lines from right to left, the behavior will be consistent.

Reply:
See below for discussion on the parsing rule.

Right now several tests rely on `PrimitiveProcessor` to strip double quotes, one test checks that it actually is an primitive and then there is that gps test where i dont understand why it even works.

Reply:
lets continue the discussion threads in the main thread below